### PR TITLE
Added Floyd's Cycle Detection Algorithm in C++, Java and Python

### PR DESCRIPTION
**Pull request by singh-shreya6**
#540 
Cycle Detection Algorithm for Linked Lists
@jainaman224 Please review

### CODE REVIEW

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    Node* next; 
+};
+Node *head = NULL;
+
+void insert(int value) 
+{ 
+    // Create a new node and adjusting links
+    Node* temp = new Node;
+    temp->data = value; 
+    temp->next = head;
```
Comment (gupta2140) [inline Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp]: You have values array as 10, 20, 30, 40. But wouldn't this function create the linked list in reverse manner? (40 -> 30 -> 20 -> 10). Can you maintain the same order as the original array?

Comment (singh-shreya6) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp]: I have made the required changes. Please review.

**Comment by jainaman224** [comment]
@gupta2140 I am leaving this PR to you.

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    for(int i= n-1; i>=0 ;i--)
+        insert(values[i]);
+  
+    // Creating a loop
+    head->next->next->next->next = head->next;
+    if(detectCycle())
+    cout<<"Cycle found"<<endl;
+    else
+    cout<<"No cycle found"<<endl;  
+    return 0;
```
Comment (gupta2140) [inline Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp]: Indentation of statements in if-else block.

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    temp->next = head; 
+    head = temp; 
+} 
+
+ // Function to detect cycle 
+bool detectCycle() 
+{ 
+    Node *slow_ptr = head;
+    Node *fast_ptr = head;   
+    while (slow_ptr!= NULL && fast_ptr!=NULL && fast_ptr->next != NULL)
```
Comment (gupta2140) [inline Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp]: no need for the unnecessary condition (slow_ptr!= NULL). fast_ptr will already take care of it.

Comment (singh-shreya6) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp]: Removed redundant checks @gupta2140

File: Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp (hunk 2 above)
Comment (singh-shreya6) [reply Floyd's_Cycle_Detection_Algorithm/Floyd's_Cycle_Detection.cpp]: Fixed Indentation @gupta2140

**Comment by gupta2140** [comment]
LGTM :+1:

**Comment by rishabhgarg25699** [comment]
Good Work .

**Pull request merged**